File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int[] freq = new int[2001];
        for (int nu : nums) {
            freq[nu+1000]++;
        }

        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> Integer.compare(b[1], a[1]));
        for (int i = 0; i < 2001; i++) {
            if (freq[i] != 0) {
                pq.add(new int[]{i-1000, freq[i]});
            }
        }
        int[] ans = new int[k];
        for (int i = 0; i < k; i++) {
            ans[i] = pq.poll()[0];
        }
        return ans;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.go
func productExceptSelf(nums []int) []int {
	n := len(nums)
	lr := make([]int, n)
	rl := make([]int, n)
	ans := make([]int, n)
	lr[0] = 1
	rl[n-1] = 1
	for i := 1; i < n; i++ {
		lr[i] = lr[i-1] * nums[i-1]
		rl[n-i-1] = rl[n-i] * nums[n-i]
	}

	for i := range n {
		if i == 0 {
			ans[i] = rl[0]
		} else if i == n -1 {
			ans[i] = lr[n-1]
		} else {
			ans[i] = lr[i] * rl[i]
		}
	}
	return ans
}
